### PR TITLE
feat: allow docker to connect via tcp

### PR DESCRIPTION
**Pull request by asnapper**
I recently started hacking on abstruse, but on my development machine i can only communicate with docker via TCP. i figured TCP support would also benefit others :-)

This PR simply parses DOCKER_HOST environment variable and configures the docker client accordingly

### CODE REVIEW

File: src/api/docker.ts [hunk 1]
```diff
@@ -11,7 +11,26 @@ import { platform } from 'os';
...
 import { demuxStream } from './utils';
 
-export const docker = new dockerode();
+export function getDockerOptions(): dockerode.DockerOptions {
+  const dockerOptions: dockerode.DockerOptions = {};
+
+  if (process.env.DOCKER_HOST) {
+    // parse docker URI in environment variable DOCKER_HOST="tcp://0.0.0.0:2375"
+    const re = /tcp\:\/\/([\w|\.]+)(\:?(\d+))/gi;
+    const [ , host, , port ] = re.exec(process.env.DOCKER_HOST)
```
[the remark this answers is not in the document]
Comment (asnapper) [reply src/api/docker.ts]: usually the whole string, but i don't want it to be deconstructed ;)

File: src/api/docker.ts [hunk 2]
```diff
@@ -11,7 +11,26 @@ import { platform } from 'os';
 import * as commandExists from 'command-exists';
 import { demuxStream } from './utils';
 
-export const docker = new dockerode();
+export function getDockerOptions(): dockerode.DockerOptions {
+  const dockerOptions: dockerode.DockerOptions = {};
+
+  if (process.env.DOCKER_HOST) {
+    // parse docker URI in environment variable DOCKER_HOST="tcp://0.0.0.0:2375"
+    const re = /tcp\:\/\/([\w|\.]+)(\:?(\d+))/gi;
```
[the remark this answers is not in the document]
Comment (asnapper) [reply src/api/docker.ts]: good call . i'll update the PR

File: src/api/docker.ts [hunk 3]
```diff
@@ -11,7 +11,26 @@ import { platform } from 'os';
...
+      dockerOptions.port = port;
+    }
+  }
+
+  return dockerOptions
+}
+
+export const docker = new dockerode(getDockerOptions());
+
 const binds = platform() === 'darwin' ? [] : ['/var/run/docker.sock:/var/run/docker.sock'];
```
Comment (asnapper) [inline src/api/docker.ts]: @jkuri i didnt fully understand what ``binds`` does, but i assume binds should be different when using docker via tcp

**Comment by irmana** [comment]
@asnapper great what you are doing, please squash your commits once you are done to follow our style guide, i.e. `feat: allow docker to connect via tcp`

**Comment by asnapper** [comment]
> squash your commits once you are done to follow our style guide for commit messages, i.e. `feat: allow docker to connect via tcp`

ah, your are talking about the commits in this pr (didn't have a coffee for a while :). don't you use 'squash merge' for pull requests? at least for such a tiny change that would make sense IMO

**Comment by irmana** [comment]
> > squash your commits once you are done to follow our style guide for commit messages, i.e. `feat: allow docker to connect via tcp`
> 
> @irmana ah, your are talking about the commits in this pr (didn't have a coffee for a while :).. don't you use 'squash merge' for pull requests? at least for such a tiny change that would make sense IMO

We're old school. ;)

**Comment by jkuri** [comment]
Closing in favor of v2

**Pull request closed without merging**